### PR TITLE
Replaced modulo with faster binary and

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**

### CODE REVIEW

**Comment by MariusVanDerWijden** [comment]
yes! currently PROGPOW_CACHE_WORDS is 512, so its a power of two. If we want to keep the modulo it should be a prime. However, I think that asics can speed up the modulo operation

File: libprogpow/ProgPow.cpp [hunk 1]
```diff
@@ -136,16 +136,16 @@ std::string ProgPow::getKern(uint64_t block_number, int dagelms, kernel_t kern)
     // load
     //ret << "// global load\n";
     if (kern == KERNEL_CUDA)
-        ret << "offset = SHFL(mix[0], loop % PROGPOW_LANES, PROGPOW_LANES);\n";
+        ret << "offset = SHFL(mix[0], loop & (PROGPOW_LANES-1), PROGPOW_LANES);\n";
```
Comment (jean-m-cyr) [inline libprogpow/ProgPow.cpp]: This fine, but all compilers will automatically replace a modulo operation with an operand that is a power of 2 with an equivalent faster and operation.

**Comment by jean-m-cyr** [comment]
Ah, I assumed these were all uints

**Comment by StefanOberhumer** [comment]
> Nice but we must assume the constraint that PROGPOW_LANES and PROGPOW_CACHE_WORDS 

Should we add a `static_assert(bool_constexpr , message )` to ensure this



)

**Pull request merged**